pkg/controller/istio/istio_controller.go

```diff
@@ -0,0 +1,348 @@
+	istioCfg := &render.IstioConfig{
+		Installation:   installation,
+		Istio:          istio,
+		IstioNamespace: render.IstioNamespace,
+		PullSecrets:    pullSecrets,
+	}
+	istioComponent := render.NewIstioComponent(istioCfg)
+
+	// Apply the image set
+	if err = imageset.ApplyImageSet(ctx, r.Client, installation.Variant, istioComponent); err != nil {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error with ImageSet", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	// Produce Helm templates for Istio
+	baseOpts := map[string]interface{}{
+		"global": map[string]interface{}{
+			"istioNamespace": render.IstioNamespace,
+		},
+	}
+	istiodOpts := map[string]interface{}{
+		"image": istioComponent.IstioPilotImage,
+		"global": map[string]interface{}{
+			"istioNamespace":         render.IstioNamespace,
+			"operatorManageWebhooks": true,
+			"proxy": map[string]interface{}{
+				"image": istioComponent.IstioProxyv2Image,
+			},
+			"proxy_init": map[string]interface{}{
+				"image": istioComponent.IstioProxyv2Image,
+			},
+		},
+		"profile": "ambient",
+	}
+	cniOpts := map[string]interface{}{
+		"image": istioComponent.IstioInstallCNIImage,
+		"global": map[string]interface{}{
+			"istioNamespace": render.IstioNamespace,
+		},
+		"ambient": map[string]interface{}{
+			"enabled":                    true,
+			"reconcileIptablesOnStartup": true,
+		},
+	}
+	if installation.KubernetesProvider == operatorv1.ProviderGKE {
+		cniOpts["global"].(map[string]interface{})["platform"] = "gke"
+	}
+	ztunnelOpts := map[string]interface{}{
+		"image": istioComponent.IstioZtunnelImage,
+		"global": map[string]interface{}{
+			"istioNamespace": render.IstioNamespace,
+		},
+	}
+	istioCfg.Resources, err = renderistio.GetResources(render.IstioNamespace, render.IstioReleaseName, baseOpts,
+		istiodOpts, cniOpts, ztunnelOpts)
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceCreateError, "Error rendering Istio resources", err, log)
+		return reconcile.Result{}, err
+	}
+
+	// Deploy Istio components
+	err = utils.NewComponentHandler(log, r, r.scheme, istio).CreateOrUpdateOrDelete(ctx, istioComponent, r.status)
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceCreateError, "Error rendering Istio resources", err, log)
+		return reconcile.Result{}, err
+	}
+
+	// Check all components are ready
+	istiodDep := &appsv1.Deployment{}
+	if err = r.Get(ctx, client.ObjectKey{Namespace: render.IstioNamespace,
+		Name: render.IstioIstiodDeploymentName}, istiodDep); err != nil {
+
+		r.status.SetDegraded(operatorv1.ResourceNotFound, "Istiod deployment not found", err, log)
+		return reconcile.Result{}, err
+	}
+	if istiodDep.Spec.Replicas == nil || istiodDep.Status.ReadyReplicas != *istiodDep.Spec.Replicas {
+		r.status.SetDegraded(operatorv1.ResourceNotReady, "Istiod deployment not ready", nil, log)
+		return reconcile.Result{RequeueAfter: time.Second * 5}, nil
+	}
+
+	istioCniDs := &appsv1.DaemonSet{}
```

Comment:
```suggestion
	istioCNIDS := &appsv1.DaemonSet{}
```

^ Acronyms should be same case.